lib/Runtime/Types/DictionaryTypeHandler.cpp

```diff
@@ -2742,7 +2742,7 @@ namespace Js
             //
 
             Js::PropertyId pid = iter.CurrentKey()->GetPropertyId();
-            if(!DynamicTypeHandler::ShouldMarkPropertyId_TTD(pid) | (!descriptor.IsInitialized) | (descriptor.Attributes & PropertyDeleted))
+            if(!DynamicTypeHandler::ShouldMarkPropertyId_TTD(pid) || (!descriptor.IsInitialized) || (descriptor.Attributes & PropertyDeleted))
```

Comment:
This is a very hot loop for the TTD snapshot (we run this over every object in the heap on every snapshot). So, I wanted to avoid the extra branching required by "||" and let the compiler optimize the bit masking/logic ops with "|" as straight-line code.